src/utils/helpers/GraphHelpers.ts

```diff
@@ -86,36 +71,149 @@ export function selectHighLightedNodes() {
+}
+
+export function selectDetailsButtonText() {
+    return selectDetailsButtonWrapper().select('text');
 }
 
 export function centerScreenToDimension(dimension: ReturnType<typeof findGroupBackgroundDimension>, scale?: number) {
     if (dimension) {
-        const svgContainer = select('#container');
+        const svgContainer = select(Selectors.CONTAINER);
 
         const width = Number(svgContainer.attr('width'));
         const height = Number(svgContainer.attr('height'));
 
-        const scaleValue = Math.min(1.3, 0.9 / Math.max(dimension.width / width, dimension.height / height));
+        const scaleValue = scale || Math.min(1.3, 0.9 / Math.max(dimension.width / width, dimension.height / height));
 
         svgContainer
+            .attr('data-scale', scaleValue)
             .transition()
-            .duration(750)
+            .duration(TRANSITION_DURATION)
             .call(
                 zoom<any, any>().on('zoom', zoomed).transform,
                 zoomIdentity
                     .translate(width / 2, height / 2)
-                    .scale(scale || scaleValue)
+                    .scale(scaleValue)
                     .translate(-dimension.x - dimension.width / 2, -dimension.y - dimension.height / 2)
             );
     }
 }
 
+function getScaleValue() {
+    return select(Selectors.CONTAINER).attr('data-scale');
+}
+
+function removeHighlightBackground() {
+    const detailsButtonRectSelection = selectDetailsButtonRect();
+    const detailsButtonTextSelection = selectDetailsButtonText();
+    selectAll([selectHighlightBackground().node(), detailsButtonRectSelection.node(), detailsButtonTextSelection.node()])
+        .transition()
+        .duration(TRANSITION_DURATION)
+        .style('opacity', 0);
+}
+
+function showHighlightBackground(dimension: ReturnType<typeof findGroupBackgroundDimension>) {
+    if (dimension) {
+        const highlightBackground = selectHighlightBackground();
+        const detailsButtonRectSelection = selectDetailsButtonRect();
+        const detailsButtonTextSelection = selectDetailsButtonText();
+
+        const scaleValue = Number(getScaleValue());
+
+        const isBackgroundNotActive = highlightBackground.style('opacity') !== String(BACKGROUND_HIGHLIGHT_OPACITY);
+
+        const scaleMultiplier = 1 / scaleValue;
+
+        const buttonWidth = 100 * scaleMultiplier;
+        const buttonHeight = 40 * scaleMultiplier;
+        const buttonMargin = 20 * scaleMultiplier;
+        const buttonX = dimension.x + dimension.width - buttonWidth - buttonMargin;
+        const buttonY = dimension.y + dimension.height - buttonHeight - buttonMargin;
+        const buttonTextFontSize = 20 * scaleMultiplier;
+        const buttonTextPositionX = dimension.x + dimension.width - buttonWidth / 2 - buttonMargin;
+        const buttonTextPositionY = dimension.y + dimension.height - buttonHeight / 2 + 7 * scaleMultiplier - buttonMargin;
+
+        if (isBackgroundNotActive) {
```

Comment:
reversing the condition and having this var named isBackgroundActive might be more readable